queue-demo-premium.tf

```
@@ -0,0 +1,157 @@
+}
+
+resource "azurerm_key_vault_secret" "envelopes_queue_listen_access_key_premium" {
+  name      = "envelopes-queue-listen-shared-access-key-premium"
+  value     = "${module.envelopes-queue-premium.primary_listen_shared_access_key}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "processed_envelopes_queue_send_access_key_premium" {
+  name      = "processed-envelopes-queue-send-shared-access-key-premium"
+  value     = "${module.processed-envelopes-queue-premium.primary_send_shared_access_key}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "processed_envelopes_queue_listen_access_key_premium" {
+  name      = "processed-envelopes-queue-listen-shared-access-key-premium"
+  value     = "${module.processed-envelopes-queue-premium.primary_listen_shared_access_key}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+
+resource "azurerm_key_vault_secret" "payments_queue_send_access_key_premium" {
+  name      = "payments-queue-send-shared-access-key-premium"
+  value     = "${module.payments-queue-premium.primary_send_shared_access_key}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "payments_queue_listen_access_key_premium" {
+  name      = "payments-queue-listen-shared-access-key-premium"
+  value     = "${module.payments-queue-premium.primary_listen_shared_access_key}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+# endregion
+
+# region connection strings and other shared queue information as Key Vault secrets
+resource "azurerm_key_vault_secret" "envelopes_queue_send_conn_str_premium" {
+  name      = "envelopes-queue-send-connection-string-premium"
+  value     = "${module.envelopes-queue-premium.primary_send_connection_string}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "envelopes_queue_listen_conn_str_premium" {
+  name      = "envelopes-queue-listen-connection-string-premium"
+  value     = "${module.envelopes-queue-premium.primary_listen_connection_string}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "envelopes_queue_max_delivery_count_premium" {
+  name      = "envelopes-queue-max-delivery-count-premium"
+  value     = "${var.envelope_queue_max_delivery_count}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "processed_envelopes_queue_send_conn_str_premium" {
+  name      = "processed-envelopes-queue-send-connection-string-premium"
+  value     = "${module.processed-envelopes-queue-premium.primary_send_connection_string}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "processed_envelopes_queue_listen_conn_str_premium" {
+  name      = "processed-envelopes-queue-listen-connection-string-premium"
+  value     = "${module.processed-envelopes-queue-premium.primary_listen_connection_string}"
```

can you fix the 'interpolation only string' deprecation warnings throughout in your change please

ideally filing a separate PR before hand to fix any other issues in the repo

```suggestion
  value     = module.processed-envelopes-queue-premium.primary_listen_connection_string
```

---

See https://github.com/hmcts/bulk-scan-shared-infrastructure/pull/255